content/en/serverless/distributed_tracing/serverless_trace_propagation.md

```diff
@@ -0,0 +1,216 @@
+- Extract the trace context in the consumer Lambda function.
+
+## Passing Trace Context to Outgoing Events
+
+The following code samples outline how to pass trace context in outgoing events to some technologies in Node and Python:
+
+{{< tabs >}}
+{{% tab "Python" %}}
+
+In Python, you can use the `get_dd_trace_context` helper function to easily pass tracing context to outgoing events:
+
+```py
+import json
+import boto3
+import os
+
+from datadog_lambda.tracing import get_dd_trace_context  # Datadog tracing helper function
+
+SNS_CLIENT = boto3.client('sns')
+
+def handler(event, context):
+    SNS_CLIENT.publish(
+        TopicArn = os.getenv('SNS_ARN'),
+        Message = 'Asynchronously invoking a Lambda function with SNS.',
+        MessageAttributes = {
+            '_datadog': {
+                'DataType': 'String',
+                'StringValue': json.dumps(get_dd_trace_context()) # Includes trace context in outgoing message to SNS topic.
+            },
+        },
+    )
+```
+
+{{% /tab %}}
+{{% tab "Node.js" %}}
+
+In Node, you can use the `getTraceHeaders` helper function to easily pass tracing context to outgoing events:
+
+```js
+const AWS = require('aws-sdk');
+const { getTraceHeaders } = require("datadog-lambda-js"); // Datadog tracing helper function
+
+const sns = new AWS.SNS();
+const eventBridge = new AWS.EventBridge();
+
+module.exports.handler = async event => {
+  const _datadog = getTraceHeaders(); // Captures current Datadog trace context.
+
+  var snsParams = {
+    Message: JSON.stringify({ data: 'sns' }),
+    MessageAttributes: {
+      '_datadog': {
+        DataType: 'String',
+        StringValue: JSON.stringify(_datadog)  // Includes trace context in outgoing message to SNS topic.
+      },
+    },
+    TopicArn: process.env.SNS_ARN
+  };
+
+  var eventBridgeParams = {
+    Entries: [{
+      EventBusName: 'tracing-tests',
+      Source: 'tracing-tests-node',
+      DetailType: 'transaction',
+      Detail: JSON.stringify({ data: 'eventBridge', _datadog }), // Includes trace context in outgoing message to event bus.
+    }]
+  };
+  
+  await sns.publish(snsParams).promise();
+  await eventBridge.putEvents(eventBridgeParams).promise()
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+## Extracting Trace Context from Consumer Lambda Functions
+
+To extract the above trace context from the consumer Lambda function, you need to define an extractor function that runs captures trace context before the execution of your Lambda function handler. To do this, configure the `DD_TRACE_EXTRACTOR` environment variable to point to the location of your extractor function. We recommend you place your extractor methods in the same file as your Lambda function handler, or all in one file, as extractors can be re-used across multiple Lambda functions. These extractors are completely customizable to fit any use case, and we included some sample extractors for common use cases in the next section.
```

Comment:
I recommend people put all extractor functions in its own file, which is a good practice (single responsibility principle). Also defining them in the lambda function file can potentially cause circular import problem in some extreme cases.